### PR TITLE
[feature](merge-cloud) Add `CloudClusterChecker` to sync cluster info…

### PR DESCRIPTION
**Pull request by deardeng**
… in cloud

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java [hunk 1]
```diff
@@ -1155,4 +1155,18 @@ public void setInstanceStatus(InstanceInfoPB.Status instanceStatus) {
         }
         this.instanceStatus = instanceStatus;
     }
+
+    public Cloud.GetClusterResponse getCloudCluster(String clusterName, String clusterId, String userName) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/system/SystemInfoService.java:1159]: move it to cloudsysteminfo if it is only used in cloud mode

File: fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudEnv.java [hunk 2]
```diff
@@ -18,20 +18,327 @@
...
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class CloudEnv extends Env {
 
     private static final Logger LOG = LogManager.getLogger(CloudEnv.class);
 
+    private CloudClusterChecker cloudClusterCheck;
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudEnv.java:50]: consider put a cloudsysteminfo object into CloudEnv and CloudClusterChecker

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by dataroaring** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**